File: zeno/debug_lexer.rs
use std::println;
fn main() {
    let input = "!-/*5;";
    let mut lexer = zeno::lexer::Lexer::new(input);
    println!("Debugging input: {}", input);
    loop {
        let tok = lexer.next_token();
        println!("Token: {:?}", tok);
        if matches!(tok, zeno::lexer::Token::Eof) {
            break;
        }
    }
}

File: debug_lexer.go
package main

import (
	"github.com/linkalls/zeno-lang/lexer"
	"github.com/linkalls/zeno-lang/token"
)

func main() {
	input := `import {type Result} from "std/result"`
	l := lexer.New(input)

	// Disabled debug loop
	for {
		tok := l.NextToken()
		if tok.Type == token.EOF {
			break
		}
	}
}
